From 487998a0aa3ec6dc37b60e32a0bd77cd9b4ca3e0 Mon Sep 17 00:00:00 2001
From: Christian Oder <myself5@carbonrom.org>
Date: Tue, 15 Jan 2019 21:59:26 +0100
Subject: [PATCH 6/6] LineageStats: explicitly cancel old jobs

* A new job gets scheduled every day or at reboot
* In case an user does not have a network connection, the jobs stack up as per AOSP documentation of setRequiredNetworkType
* We only want to report to the server once, so there is no need to keep old jobs any longer, we therefore cancel the old jobs

Change-Id: I3c6baaf3b10658aa2e49b8ac8d207aa9eac82772
---
 .../lineageos/lineageparts/lineagestats/AnonymousStats.java   | 2 +-
 .../lineageos/lineageparts/lineagestats/ReportingService.java | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/org/lineageos/lineageparts/lineagestats/AnonymousStats.java b/src/org/lineageos/lineageparts/lineagestats/AnonymousStats.java
index eff1abc..96a2555 100644
--- a/src/org/lineageos/lineageparts/lineagestats/AnonymousStats.java
+++ b/src/org/lineageos/lineageparts/lineagestats/AnonymousStats.java
@@ -49,7 +49,7 @@ public class AnonymousStats extends SettingsPreferenceFragment {
                 .commit();
     }
 
-    private static int getLastJobId(Context context) {
+    public static int getLastJobId(Context context) {
         return getPreferences(context).getInt(KEY_LAST_JOB_ID, 0);
     }
 
diff --git a/src/org/lineageos/lineageparts/lineagestats/ReportingService.java b/src/org/lineageos/lineageparts/lineagestats/ReportingService.java
index 473e76b..e7632ea 100644
--- a/src/org/lineageos/lineageparts/lineagestats/ReportingService.java
+++ b/src/org/lineageos/lineageparts/lineagestats/ReportingService.java
@@ -45,6 +45,7 @@ public class ReportingService extends IntentService {
         String deviceCarrier = Utilities.getCarrier(getApplicationContext());
         String deviceCarrierId = Utilities.getCarrierId(getApplicationContext());
 
+        final int lineageOldJobId = AnonymousStats.getLastJobId(getApplicationContext());
         final int lineageOrgJobId = AnonymousStats.getNextJobId(getApplicationContext());
 
         if (DEBUG) Log.d(TAG, "scheduling job id: " + lineageOrgJobId);
@@ -71,6 +72,9 @@ public class ReportingService extends IntentService {
                 .setPersisted(true)
                 .build());
 
+        // cancel old job in case it didn't run yet
+        js.cancel(lineageOldJobId);
+
         // reschedule
         AnonymousStats.updateLastSynced(this);
         ReportingServiceManager.setAlarm(this);
-- 
2.17.1

